src/BasicAccessFactory.php

```diff
@@ -29,6 +30,7 @@ public function __invoke(ContainerInterface $container) : BasicAccess
             );
         }
 
+        /** @var string|null $realm */
```

Comment:
You might want to require webmozart/assert and use Assert::nullOrString() instead of inline annotation 👍

Reply:
It will be BC change, isn't it?

`webmozart/assert` will throw `\InvalidArgumentException`
In current code, we throw own package exception `Mezzio\Authentication\Exception\InvalidConfigException`

Reply:
Similar to above feedback: right now we **assume** `null|string`, but so many things can go wrong if it is not a `string` - instead, an exception is already a good improvement